src/main/java/org/apache/sling/resource/collection/ResourceCollection.java

```diff
@@ -67,7 +68,16 @@ public interface ResourceCollection {
     /**
      * Returns additional properties for a particular resource in Collection entry.
      *
-     * @return properties of the Collection entry as <code>ModifiableValueMap</code>, returns null if entry found.
+     * @return properties of the Collection entry as the specified type, returns empty if no entry found or 
+     *  the entry cannot be adapted to the type
+     */
+    <T> Optional<T> getPropertiesAs(Resource resource, Class<T> type);

```

Comment:
You can provide as type also a SlingModel class. Is this something we want to support?

Reply:
@joerghoh yeah, I debated making it more specific (e.g. get the `ValueMap` or `Resource`) but decided allowing any adaptable target class would both make the code cleaner and be more generic.

Reply:
I wonder if we need that level of "generic" at this point here at all.

Reply:
Good point, especially in light of this class not functionally changing in a decade. I reverted to just returning the ValueMap and ensuring that the ValueMap will never be null (though it may be empty if the resource is not found in the collection or the adaption fails)